An article explaining the importance of do-while(0) loop

sirilcodes opened the pull request:
Added a new folder `do-while-magic`. 
- Mainly focuses on why do-while is important.
- It is written from a FPP perspective, coz I want to connect with the reader.
- I kept it as short and informative as I can, but I am open to feedback
- The reader would get a glimpse of best practices  while writing C code.

sirilcodes committed:
created a new dir

sirilcodes committed:
initial draft

sirilcodes committed:
Added article related to do-while(0)
- Explains the importance of do-while
- I have written it in fpp. I want to talk to the reader.
- Article assumes the reader understands Macros
- Its a 10 min read.

streetdogg commented on do-while-magic/index.md:
```diff
@@ -0,0 +1,177 @@
+# replace the "sample/cover.png" with path to a 16:9 image
+# this image will be shown in SEO optimization and when
+# you share the post on social media.
+thumbnail: "./do_while_title.png"
```
Please align this path of this as recommended in [documentation for images](https://github.com/pyjamabrah/pyjamabrah-community-contributions?tab=readme-ov-file#images)

streetdogg commented on do-while-magic/index.md:
```diff
@@ -0,0 +1,177 @@
+thumbnail: "./do_while_title.png"
+
+# Add your GitHub handle here
+author: "github.com/sirilcodes"
```
`github.com/` can be excluded. Only username :)

sirilcodes replied:
Updated as per the comment

streetdogg commented on do-while-magic/do_while_title.png:
This file can be named, `thumbnail.png` or `cover.png`. Also, would recommend making the font bigger so the thumbnail becomes even m ore readable :)

streetdogg commented on do-while-magic/index.md:
````diff
@@ -0,0 +1,177 @@
+
+Here's a complete example showing proper macro usage:
+
+```c
````
If there is an excepted output for this. Please add that as well.

streetdogg commented:
> Added a new folder `do-while-magic`.
> 
> * Mainly focuses on why do-while is important.
> * It is written from a FPP perspective, coz I want to connect with the reader.
> * I kept it as short and informative as I can, but I am open to feedback
> * The reader would get a glimpse of best practices  while writing C code.

The article looks good to me. Thank you for the contribution :)

Left few comments to align the format with the document parser. Also, is the content taken from a LLM? If so, please be sure to check for the accuracy and be sure to try the examples. Attaching the output in as part of the article further improves the quality.

streetdogg reviewed (changes requested):
Added few comments

sirilcodes committed:
Resolved Comments
- fixed the path for cover
- github username added
- Code snippet with output added
- used LLMs for grammer and typos.

sirilcodes commented:
Thanks for the review Piyush. I made changes as per the above comments. Please review and let me know if any additional changes required.

sirilcodes committed:
Changes in the code snippets
- previous code was not accurately demonstrating the
do-while loop.

sirilcodes commented:
I made changes in the code snippet to properly demonstrate the use of do while. Please consider the above commit for review. Let me know if any changes required. Thanks :)

streetdogg reviewed (approved):
This looks good. Thank you :)
